### PR TITLE
Added new wifi widget using iw

### PR DESCRIPTION
**Pull request by Mellich**
Sorry for the inconvenience, now the second try and only my additions:
On the Arch Linux Wiki pages the package 'wireless_tools' containing iwconfig is listed as deprecated.
https://wiki.archlinux.org/index.php/Wireless_network_configuration#Manual_setup
So I added a new widget to collect wifi information using iw instead.

### CODE REVIEW

File: widgets/wifiiw.lua [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    winfo["{ssid}"] =  -- SSID can have almost anything in it until new line
+      string.match(iwresult, 'SSID: ([^\n]*)') or winfo["{ssid}"]
+    winfo["{mode}"] =  -- everything after 'type ' until new line
+      string.match(iwresult, "type ([^\n]*)") or winfo["{mode}"]
+    winfo["{chan}"] =  -- Channels are plain digits
+      tonumber(string.match(iwresult, "channel ([%d]+)") or winfo["{chan}"])
+    winfo["{rate}"] =  -- We don't want to display Mb/s
+      tonumber(string.match(iwresult, "tx bitrate: ([%d%.]*)") or winfo["{rate}"])
+    winfo["{freq}"] =  -- Frequency are plain digits
+      tonumber(string.match(iwresult, "freq: ([%d]+)") or winfo["{link}"])
```
Comment (Mic92) [inline widgets/wifiiw.lua]: `{link}` is never initialized here.

File: widgets/wifiiw.lua [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            if f then
+                iw = p .. "/iw"
+                f:close()
+                break
+            end
+        end
+    end
+
+    -- Get data from iw where available
+    local f = io.popen(iw .." dev ".. warg .. " link 2>&1;" .. iw .." dev ".. warg .. " info 2>&1")
```
Comment (Mic92) [inline widgets/wifiiw.lua]: actually path search could be also delegated to the shell, which does also check if the file is executable.

```
io.popen("export PATH=$PATH:/sbin/:/usr/sbin:/usr/local/sbin; iw dev " ...
```

In that case the loop above, can be removed.

Also please escape shell characters from warg using `helpers.shellquote(warg)`.

File: widgets/wifiiw.lua [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+local iwcpaths = { "/sbin", "/usr/sbin", "/usr/local/sbin", "/usr/bin" }
+-- }}}
+
+
+-- {{{ Wireless widget type
+local function worker(format, warg)
+    if not warg then return end
+
+    -- Default values
+    local winfo = {
```
Comment (Mic92) [inline widgets/wifiiw.lua]: When I think about, we could also add an empty table here:

```
local winfo = {}
```

and then assign default values to fields like this:

```lua
winfo["{ssid}"] =  -- SSID can have almost anything in it until new line
    string.match(iwresult, 'SSID: ([^\n]*)') or "N/A"
```

File: widgets/wifiiw.lua [hunk 4]
```diff
@@ -0,0 +1,89 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2010, Adrian C. <anrxc@sysphere.org>
```
Comment (Mic92) [inline widgets/wifiiw.lua]: I don't think Adrian did this :)

File: widgets/wifiiw.lua (hunk 1 above)
Comment (Mellich) [inline widgets/wifiiw.lua]: actually there is no output of iw that equals the link quality provided by iwconfig. Or at least I didn't find it. Nevertheless, I added a {linp} field which calculates the link quality using the signal level.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**